theano/compile/function_module.py

```diff
@@ -963,11 +971,18 @@ def __call__(self, *args, **kwargs):
 
                 assert len(self.output_keys) == len(outputs)
 
-                return dict(izip(self.output_keys, outputs))
+                if output_subset is None:
+                    return dict(izip(self.output_keys, outputs))
+                else:
+                    return dict((self.output_keys[index], outputs[index])
+                                for index in output_subset)
 
             if output_subset is None:
                 return outputs
             else:
+                if not all(isinstance(index, int) for index in output_subset):
```

Comment:
I would not do that. It will cause extra overhead to good call. Bad call will still give error that I think is clear enough as we will see the code that is a one liner.